sort/fusion.py

```diff
@@ -4,4 +4,26 @@
 
 
 def sort(array: list[int]) -> list[int]:
+# Je vérifie que la liste possède plus d'un élément pour engager la division de liste
+    if not len(array) <= 1:
+        f_half: list[int] = array[:int(len(array)/2)]
+        s_half: list[int] = array[int(len(array)/2):]
+# Je répète l'opération jusqu'à n'avoir qu'une valeur dans chaque tableau
+        return merge(sort(f_half), sort(s_half))
+    else:
+        return array
+
+
+# Fonction de fusion de liste avec tri par croissance
+def merge(merge_array: list[int], array: list[int]) -> list[int]:
+    for value in merge_array:
+        too_big: bool = True
+        for index in range(len(array)):
+            if value < array[index]:
+                too_big = False
+                array.insert(index, value)
+                break
```

Comment:
Là, tu abandonnes le principe du tri par fusion pour faire une sorte de tri par insertion.
Je t'invite à prendre le temps de bien comprendre l'algo fusion avec le schéma. 
Le problème, c'est que cette méthode dégrade la qualité de l'algorithme : d'une simple boucle while dans le tri par fusion, on passe ici à une double boucle (for dans for).
On a donc une complexité O(n² log n) au lieu de O(n log n), et ce carré va rendre les calculs extrèmement longs sur de grandes entrées.
Donc attention, les tests automatiques ne font pas tout : ils passent, mais l'implémentation n'est pas correcte et la complexité devient même supérieur au tri par sélection et insertion.
A dispo pour expliquer cela plus en profondeur bien sûr.